src/app/games/quebec-castles/QuebecCastlesRules.ts

```diff
@@ -0,0 +1,686 @@
+import { MGPFallible, MGPOptional, MGPValidation, Utils } from '@everyboard/lib';
+
+import { GameNode } from 'src/app/jscaip/AI/GameNode';
+import { QuebecCastlesDrop, QuebecCastlesMove, QuebecCastlesTranslation } from './QuebecCastlesMove';
+import { QuebecCastlesState } from './QuebecCastlesState';
+import { GameStatus } from 'src/app/jscaip/GameStatus';
+import { ConfigurableRules } from 'src/app/jscaip/Rules';
+import { Table, TableUtils } from 'src/app/jscaip/TableUtils';
+import { BooleanConfig, EnumConfig, NumberConfig, RulesConfigDescription, RulesConfigDescriptionLocalizable } from 'src/app/components/wrapper-components/rules-configuration/RulesConfigDescription';
+import { MGPValidators } from 'src/app/utils/MGPValidator';
+import { Coord, CoordFailure } from 'src/app/jscaip/Coord';
+import { Player, PlayerOrNone } from 'src/app/jscaip/Player';
+import { PlayerMap } from 'src/app/jscaip/PlayerMap';
+import { RulesFailure } from 'src/app/jscaip/RulesFailure';
+import { MoveCoordToCoord } from 'src/app/jscaip/MoveCoordToCoord';
+import { DirectionFailure } from 'src/app/jscaip/Direction';
+import { Ordinal } from 'src/app/jscaip/Ordinal';
+import { Localized } from 'src/app/utils/LocaleUtils';
+
+export class QuebecCastlesFailure {
+
+    public static readonly INVALID_INVADER_DISTANCE: (distance: number) => string = (distance: number) => $localize`Move distance must be 2 for invader, not ${ distance }`;
+
+    public static readonly INVALID_DEFENDER_DISTANCE: (distance: number) => string = (distance: number) => $localize`Move distance must be 1 for defender, not ${ distance }`;
+
+    public static readonly MUST_DROP_IN_YOUR_TERRITORY: Localized = () => $localize`Must drop in your own territory`;
```

Comment:
Voir comment uniformiser avec la structure de nos autres messages. Ici on parle clairement à l'utilisateur ("your own territory"), alors faut faire une phrase complète ("You must drop [...]", pas juste "Must drop").
Les autres messages sont à voir, "Cannot drop in move phase" semble faire sens, mais est-ce que c'est uniforme avec nos autres messages d'autres jeux ?